### PR TITLE
add DisactivateUserEducationConsumer

### PR DESCRIPTION
**Pull request by borovHokage**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/987

### CODE REVIEW

File: src/EducationService.Data/UserRepository.cs [hunk 1]
```diff
@@ -33,5 +31,33 @@ await _provider.UsersEducations
           .Where(uc => uc.UserId == userId)
           .ToListAsync());
     }
+
+    public async Task<bool> DisactivateCertificateAndEducations(Guid userId, Guid modifiedBy)
```
Comment (MariyaSin) [inline src/EducationService.Data/UserRepository.cs]: no need return bool here

File: src/EducationService.Models.Dto/Configurations/RabbitMqConfig.cs [hunk 2]
```diff
@@ -9,6 +9,7 @@ public class RabbitMqConfig : BaseRabbitMqConfig
   {
     public string GetUserEducationsEndpoint { get; set; }
     public string GetUserSkillsEndpoint { get; set; }
+    public string DisactivateUserEducationEndpoint { get; set; }
```
Comment (MariyaSin) [inline src/EducationService.Models.Dto/Configurations/RabbitMqConfig.cs]: ...UserEducations...

File: src/EducationService/Startup.cs [hunk 3]
```diff
@@ -110,6 +111,11 @@ private void ConfigureConsumers(IServiceCollectionBusConfigurator x)
       {
         ep.ConfigureConsumer<GetUserSkillsConsumer>(context);
       });
+
+      cfg.ReceiveEndpoint(rabbitMqConfig.DisactivateUserEducationEndpoint, ep =>
```
Comment (MariyaSin) [inline src/EducationService/Startup.cs]: ...UserEducations..

File: src/EducationService/Startup.cs [hunk 4]
```diff
@@ -110,6 +111,11 @@ private void ConfigureConsumers(IServiceCollectionBusConfigurator x)
...
+      {
+        ep.ConfigureConsumer<DisactivateUserEducationConsumer>(context);
```
Comment (MariyaSin) [inline src/EducationService/Startup.cs]: ...UserEducations.....

File: src/EducationService/appsettings.json [hunk 5]
```diff
@@ -37,6 +37,7 @@
     //Education
     "GetUserEducationsEndpoint": "EducationService_GetUserEducations",
     "GetUserSkillsEndpoint": "EducationService_GetUserSkills",
+    "DisactivateUserEducationEndpoint": "EducationService_DisactivateUser",
```
Comment (MariyaSin) [inline src/EducationService/appsettings.json]: ..UserEducations..

**Review by MariyaSin — APPROVED** [review]

File: src/EducationService.Data/Interfaces/IUserRepository.cs [hunk 6]
```diff
@@ -11,5 +11,7 @@ public interface IUserRepository
   {
     Task<(List<DbUserCertificate> userCertificates, List<DbUserEducation> userEducations)>
       GetAsync(Guid userId);
+
+    Task<bool> DisactivateCertificateAndEducations(Guid userId, Guid modifiedBy);
```
Comment (ndoriomedov) [inline src/EducationService.Data/Interfaces/IUserRepository.cs]: add Async to name's end

File: src/EducationService/appsettings.json (hunk 5 above)
Comment (borovHokage) [reply src/EducationService/appsettings.json]: done

File: src/EducationService/Startup.cs (hunk 4 above)
Comment (borovHokage) [reply src/EducationService/Startup.cs]: done

File: src/EducationService.Models.Dto/Configurations/RabbitMqConfig.cs (hunk 2 above)
Comment (borovHokage) [reply src/EducationService.Models.Dto/Configurations/RabbitMqConfig.cs]: done

File: src/EducationService.Data/UserRepository.cs (hunk 1 above)
Comment (borovHokage) [reply src/EducationService.Data/UserRepository.cs]: fixed

File: src/EducationService.Data/UserEducationRepository.cs [hunk 7]
```diff
@@ -6,6 +6,8 @@
 using Microsoft.AspNetCore.JsonPatch;
 using Microsoft.EntityFrameworkCore;
 using System;
+using System.Collections.Generic;
+using System.Linq;
```
Comment (ndoriomedov) [inline src/EducationService.Data/UserEducationRepository.cs:10]: do we need this? these are the only changes to this file - didn't it work before without them?

File: src/EducationService.Data/UserRepository.cs [hunk 8]
```diff
@@ -33,5 +31,33 @@ await _provider.UsersEducations
           .Where(uc => uc.UserId == userId)
           .ToListAsync());
     }
+
+    public async Task DisactivateCertificateAndEducations(Guid userId, Guid modifiedBy)
```
Comment (ndoriomedov) [inline src/EducationService.Data/UserRepository.cs]: add Async to the end of method's name

**Review by ndoriomedov — APPROVED** [review]

File: src/EducationService.Data/UserEducationRepository.cs (hunk 7 above)
Comment (borovHokage) [reply src/EducationService.Data/UserEducationRepository.cs:10]: just useless link

File: src/EducationService.Data/Interfaces/IUserRepository.cs (hunk 6 above)
Comment (borovHokage) [reply src/EducationService.Data/Interfaces/IUserRepository.cs]: done

**Pull request merged**